recipe/build.sh

```diff
@@ -4,13 +4,19 @@ set -exv
 make generate --jobs ${CPU_COUNT}
 
 # Duplicate lists because of https://bitbucket.org/icl/magma/pull-requests/32
-export CUDA_ARCH_LIST="sm_35,sm_50,sm_60,sm_70,sm_80"
-export CUDAARCHS="35-virtual;50-virtual;60-virtual;70-virtual;80-virtual"
+export CUDA_ARCH_LIST="sm_35,sm_50,sm_60,sm_70,sm_75,sm_80"
+export CUDAARCHS="35-virtual;50-virtual;60-virtual;70-virtual;75-virtual;80-virtual"
 
 # Only build the lowest non-deprecated arch to minimize build time
 if [[ "$target_platform" == "linux-ppc64le" || "$target_platform" == "linux-aarch64" ]]; then
-  export CUDA_ARCH_LIST="sm_60"
-  export CUDAARCHS="60-virtual"
+  export CUDA_ARCH_LIST="sm_60;sm_80"
+  export CUDAARCHS="60-virtual;80-virtual"
+fi
+
+# Only build the lowest non-deprecated arch to minimize build time
+if [[ "$cuda_compiler_version" == "12.0" ]]; then
+  export CUDA_ARCH_LIST="sm_50,sm_60,sm_70,sm_75,sm_80,sm_89"
+  export CUDAARCHS="50-virtual;60-virtual;70-virtual;75-virtual;80-virtual;89-virtual"
```

Comment:
Would it make sense to add 90?

```suggestion
  export CUDA_ARCH_LIST="sm_50,sm_60,sm_70,sm_75,sm_80,sm_89,sm_90"
  export CUDAARCHS="50-virtual;60-virtual;70-virtual;75-virtual;80-virtual;89-virtual;90-virtual"
```

Reply:
No. The only real device for 90 right now is the H100, which is supercomputer hardware. Thus, outside our target demographic.

Reply:
Unless people are using conda-forge to run workloads on H100s that are rented by the hour? In my experience, supercomputers have admins which build all the popular packages for their specific machines.

Reply:
Yeah that's possible. Maybe we can raise an issue and track demand there?

Reply:
We use conda-forge to run stuff on our high-performance computer. Unfortunately we don't have H100s, but unless adding sm_90 has some drawbacks, I would be in favour of adding it :)

Reply:
Build times for this package historically have bumped against the 6 hour CI time limit, that's why the targets are all virtual instead of real and virtual. If there seems to be time, I will add 90.

Reply:
Seems like there's plenty of time to spare.